src/components/general/modals/share-experiences-modal.js

```diff
@@ -1,17 +1,43 @@
-import React, { useState } from 'react';
+import React, { useState, useContext } from 'react';
 import styled from 'styled-components';
 
-import { Dialog, DialogTitle, DialogContent, IconButton } from '@material-ui/core';
+import { Dialog, DialogTitle, IconButton } from '@material-ui/core'; 
 import CloseIcon from '@material-ui/icons/Close';
+import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
+import InitialExperiences from '../../experiences/initial-experiences'; 
+import PositiveExperiences from '../../experiences/positive-experiences';  
+import NegativeExperiences from '../../experiences/negative-experiences';  
+import ThanksExperiences from '../../experiences/thanks-experiences'; 
 
+import ExperienceService from '@services/experience.service';
 import ModalService from '@services/modal.service'; 
 
-export default function ShareExperiencesModal() {
-  const modalService = ModalService.getInstance();
-  modalService.registerModal('promptShareExperiences', toggleModal);
+import { AppContext } from '@contexts/app.context';
+
+export default function ShareExperiencesModal(props) {
+
+  const { appState } = useContext(AppContext);  
+
+  const modalService = ModalService.getInstance(); 
+  const experienceService = ExperienceService.getInstance(); 
+
+  modalService.registerModal('promptShareExperiences'+props.testTitle, toggleModal);
 
   const [open, setOpen] = useState(false);
-  const [scroll, setScroll] = useState('paper');
+  const [scroll, setScroll] = useState('paper');  
```

Comment:
try to use one state object that has all of these instead of individual variables